modules/vba_documents/lib/vba_documents/upload_validator.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require_dependency 'vba_documents/pdf_inspector'
+
+require 'central_mail/utilities'
+require 'central_mail/service'
+require 'pdf_info'
+
+module VBADocuments
+  module UploadValidations
+    include CentralMail::Utilities
+
+    def update_pdf_metadata(model, inspector)
+      model.update(uploaded_pdf: inspector.pdf_data)
+    end
+
+    def validate_parts(parts)
+      unless parts.key?(META_PART_NAME)
+        raise VBADocuments::UploadError.new(code: 'DOC102',
+                                            detail: 'No metadata part present')
+      end
+      unless parts[META_PART_NAME].is_a?(String)
+        raise VBADocuments::UploadError.new(code: 'DOC102',
+                                            detail: 'Incorrect content-type for metadata part')
+      end
+      unless parts.key?(DOC_PART_NAME)
+        raise VBADocuments::UploadError.new(code: 'DOC103',
+                                            detail: 'Submission did not include a document.')
+      end
+      if parts[DOC_PART_NAME].is_a?(String)
+        raise VBADocuments::UploadError.new(code: 'DOC103',
+                                            detail: 'Incorrect content-type for document part')
+      end
+      # TODO: validate type and sequential naming of attachment parts
```

Comment:
Is this still needed?

Reply:
Thanks for catching that.  It was done. removed todo.